skyportal/handlers/api/annotation_services.py

```diff
@@ -277,3 +280,125 @@ def post(self, obj_id):
             payload={'obj_key': obj.internal_key},
         )
         return self.success()
+
+
+class DatalabQueryHandler(BaseHandler):
+    """
+    ---
+    description: |
+        get photo(z) of nearby sources and post them as an annotation
+        based on cross-matches to some catalog (default is LegacySurvey DR8).
+    parameters:
+    - in: path
+        name: obj_id
+        required: true
+        schema:
+          type: string
+        description: ID of the object to retrieve photoz's for
+      - in: query
+        name: catalog
+        required: false
+        schema:
+          type: string
+        description: |
+          The name of the catalog key, associated with a catalog cross match,
+          from which the photoz data should be retrieved.
+          Default is ls_dr8.
+      - in: query
+        name: crossmatchRadius
+        required: false
+        schema:
+          type: number
+        description: |
+          Crossmatch radius (in arcseconds) to retrieve photoz's
+          Default is 2.
+      - in: query
+        name: group_ids
+        required: false
+        schema:
+          type: array
+          items:
+            type: integer
+        description: |
+          List of group IDs corresponding to which groups should be
+          able to view annotation. Defaults to all of requesting user's groups.
+    responses:
+      200:
+        content:
+          application/json:
+            schema: Success
+      400:
+        content:
+          application/json:
+            schema: Error
+    """
+
+    @auth_or_token
+    def post(self, obj_id):
+        obj = Obj.query.get(obj_id)
+        if obj is None:
+            return self.error('Invalid object id.')
```

Comment:
shouldn't there be some check if the Obj is accessible to the user?

Reply:
good catch.